image-loader/app/controllers/ImageLoaderController.scala

```diff
@@ -65,8 +96,122 @@ class ImageLoaderController(auth: Authentication,
+        Future.unit
+      case Right(s3data) =>
+        val s3IngestObject = S3IngestObject(s3data, store)
+        val approximateReceiveCount = getApproximateReceiveCount(sqsMessage)
+        logMarker ++ Map(
+          "uploadedBy" -> s3IngestObject.uploadedBy,
+          "uploadedTime" -> s3IngestObject.uploadTime,
+          "contentLength" -> s3IngestObject.contentLength,
+          "filename" -> s3IngestObject.filename,
+          "isUiUpload" -> s3IngestObject.maybeMediaIdFromUiUpload.isDefined,
+        )
+
+        if (approximateReceiveCount > 2) {
+          val errorMessage = s"File processing has been attempted $approximateReceiveCount times. Moving to fail bucket."
+          logger.warn(logMarker, errorMessage)
+          store.moveObjectToFailedBucket(s3IngestObject.key)
+
+          s3IngestObject.maybeMediaIdFromUiUpload match {
+            case Some(imageId) => uploadStatusTable.updateStatus( //FIXME use set status to avoid potential ConditionNotMet (when status table rows have expired/TTL)
+              imageId, UploadStatus(StatusType.Failed, Some(errorMessage))
+            )
+            case None => Future.unit
+          }
+        }
+
+        attemptToProcessIngestedFile(s3IngestObject).transformWith {
+          case Failure(exception) =>
+            logger.warn(logMarker, s"Failed to process file: ${exception.toString}. Moving to fail bucket")
+            store.moveObjectToFailedBucket(s3IngestObject.key)
+            Future.unit
+          case Success(digestedFile) =>
+            logger.info(logMarker, s"Successfully processed image ${digestedFile.file.getName}")
+            store.deleteObjectFromIngestBucket(s3IngestObject.key)
+            Future.unit
+        }
+    }
+  }
+
+
+  private def attemptToProcessIngestedFile(s3IngestObject:S3IngestObject)(implicit logMarker:LogMarker): Future[DigestedFile] = {
+
+    logger.info(logMarker, "Attempting to process file")
+    val tempFile = createTempFile("s3IngestBucketFile")
+
+    val digestedFile = downloader.download(
+      inputStream = s3IngestObject.getInputStream(),
+      tempFile,
+      expectedSize = s3IngestObject.contentLength
+    )
+    logMarker ++ Map(
+      "SHA-1" -> digestedFile.digest
+    )
+
+    val futureUploadStatusUri = uploadDigestedFileToStore(
+        digestedFileFuture = Future(digestedFile),
+        uploadedBy = s3IngestObject.uploadedBy,
+        identifiers =  None,
+        uploadTime = Some(s3IngestObject.uploadTime.toString) , // upload time as iso string - uploader uses DateTimeUtils.fromValueOrNow
+        filename = Some(s3IngestObject.filename)
+    )
+
+    // under all circumstances, remove the temp files
+    futureUploadStatusUri.onComplete { _ =>
+      Try { deleteTempFile(tempFile) }
+    }
+
+    handleUploadCompletionAndUpdateUploadStatusTable(futureUploadStatusUri, digestedFile)
+      .recover {
+        case e:Exception => Future.failed(e)
+        case _ => Future.failed(new Exception)
+      }
```

Comment:
I think you can probably drop this recover block? let the already failed future continue as is

Reply:
done